### DISCUSSION TITLE
Stay Away from Reolink Trackmix

### DISCUSSION THREAD

**Discussion by rpm5099** [blakeblackshear/frigate#15156] [General]
There is an issue with stuttering/dropped frames with the Reolink Trackmix via RTSP, in the Reolink application it is fine but obviously that does no good for Frigate.  The picture looks very good otherwise.  I have been in contact with their support for about 2 months and I've gotten nowhere.  I've sent them raw videos, verbose ffmpeg logs, and finally (after they acknowledged there is definitely an issue with the camera) a full packet capture - it's been "sent to the engineers".  I've asked to speak directly with the engineers, and was told no.  

On top of that the main stream is fixed at H265 and there is no option to change the codec.  I actually got Reolink to send me H264 firmware, and it looked so bad it was clearly unusable - [Reolink TrackMix Wifi Firmware v3.0.0.3748_2408132492 - H.264 main stream](https://drive.google.com/file/d/1-ikjmEA4hQOjCOZnFE4MLwGkT89LymlX).  They also do not hold their home setting at all, it's way off within hours of setting it - I have 2 and I've confirmed all of this on both.  

It would be one thing if they offered me a refund or an exchange for a different device, they offered me $20.  Reolink is not a brand I will use anymore if there are any equivalent options in the same price range.

Here is what I have shared with Reolink support - [stuttering](https://drive.google.com/drive/folders/1B9VNocE3kLceCD85Jcnx6f8BA23ihSoR)

**Comment by mrBobMusic** [comment]
> There is an issue with stuttering/dropped frames with the Reolink Trackmix via RTSP, in the Reolink application it is fine but obviously that does no good for Frigate. The picture looks very good otherwise. I have been in contact with their support for about 2 months and I've gotten nowhere. I've sent them raw videos, verbose ffmpeg logs, and finally (after they acknowledged there is definitely an issue with the camera) a full packet capture - it's been "sent to the engineers". I've asked to speak directly with the engineers, and was told no.
> 
> On top of that the main stream is fixed at H265 and there is no option to change the codec. I actually got Reolink to send me H264 firmware, and it looked so bad it was clearly unusable - [Reolink TrackMix Wifi Firmware v3.0.0.3748_2408132492 - H.264 main stream](https://drive.google.com/file/d/1-ikjmEA4hQOjCOZnFE4MLwGkT89LymlX). They also do not hold their home setting at all, it's way off within hours of setting it - I have 2 and I've confirmed all of this on both.
> 
> It would be one thing if they offered me a refund or an exchange for a different device, they offered me $20. Reolink is not a brand I will use anymore if there are any equivalent options in the same price range.
> 
> Here is what I have shared with Reolink support - [stuttering](https://drive.google.com/drive/folders/1B9VNocE3kLceCD85Jcnx6f8BA23ihSoR)


I have two Trackmix cameras. Today I tried to set them up in Frigate. I did not get as far as you did. I don't think I have created the correct rtsp string. 
Here's what I have:  - path: rtsp://admin:myPassWord@192.168.12.218:554/av_stream/ch0
I tried testing my link in VLC, as suggested in this video.
 https://www.youtube.com/watch?v=cRjKu0Y6USY&t=304s

VLC errors out, with "connection failed".   Could you tell me the correct syntax for the rtsp string?

Thanks...

**Comment by rpm5099** [comment]
@mrBobMusic Sure, if you go to `rtsp://admin:myPassWord@192.168.12.218:554/` it should give you the main stream by default.  You get the same stream by going to `rtsp://admin:myPassWord@192.168.12.218:554/Preview_01_main`.  The sub stream is on `rtsp://admin:myPassWord@192.168.12.218:554/Preview_01_sub`.  This isn't exactly easy to find I agree.

I would be interested to know if you have stuttering/dropped frames in your feed.

**Comment by mguebert** [comment]
I have four of the trackmix poe cameras, and in the reolink app they work perfectly. I just migrated from blue iris, where they also worked fine. But with frigate I get crashing of ffmpeg with errors related to audio especially error
2024-12-06 07:42:42
ffmpeg.Deck.audio_record
[aac @ 0x60145db4e600] Input buffer exhausted before END element found
error
2024-12-06 07:42:42
ffmpeg.Deck.audio_record
Error while decoding stream #0:1: Invalid data found when processing input
error
2024-12-06 07:42:42
ffmpeg.Deck.audio_record
[aac @ 0x60145db4e600] channel element 2.10 is not allocated
error
2024-12-06 07:42:42
ffmpeg.Deck.audio_record
Error while decoding stream #0:1: Invalid data found when processing input

It is frustrating for sure, because as I indicated they worked fine in blue iris. But I wanted to move away from BI because it is resource intensive.

**Reply by rpm5099** [reply]
You don't get the stuttering every few seconds?  Maybe it's just the wifi version (although I have them connected via ethernet).  Would you post your camera and frigate/go2rtc settings?  Anything is possible I suppose...

**Reply by BoutTime2** [reply]
I tried setting it up also and I get stuttering/dropped frames. But I put it down to the fact the Reolink Software was trying to detect/track/zoom/record at the same time and it was imply too much for the CPU and network to handle. What I should have done is disable all of these features in the Reolink App, but having given it some thought I think the Reolink Firmware is using proprietary methods to track/zoom with both lenses and Frigate is just not going to be able to do this. I had everything working on the Wide Angle lens only, both main and sub streams. So reluctantly I'll continue to use the Reolink app and FTP to back up recordings and snapshots to my Unraid server.
Home Assistant is providing excellent clickable snapshot notifications to my mobile phones for Animal, Vehicle or People detections.

**Reply by rpm5099** [reply]
I really think Reolink has failed to correctly implement the RTSP protocol in their feed, and frankly they don't really care if it forces some people to pay for their cloud service.

**Reply by borgqueenx** [reply]
> I have four of the trackmix poe cameras, and in the reolink app they work perfectly. I just migrated from blue iris, where they also worked fine. But with frigate I get crashing of ffmpeg with errors related to audio especially error 2024-12-06 07:42:42 ffmpeg.Deck.audio_record [aac @ 0x60145db4e600] Input buffer exhausted before END element found error 2024-12-06 07:42:42 ffmpeg.Deck.audio_record Error while decoding stream #0:1: Invalid data found when processing input error 2024-12-06 07:42:42 ffmpeg.Deck.audio_record [aac @ 0x60145db4e600] channel element 2.10 is not allocated error 2024-12-06 07:42:42 ffmpeg.Deck.audio_record Error while decoding stream #0:1: Invalid data found when processing input
> 
> It is frustrating for sure, because as I indicated they worked fine in blue iris. But I wanted to move away from BI because it is resource intensive.

Blue iris is always more stable ss frigate as its desktop software. Frigate runs in a browser. Just like vlc will also work eith reolink and other streams, frigate is much more sensitive.

**Reply by rpm5099** [reply]
That was not the problem with the Reolink TrackMix cameras that I had, you must have a different hardware/firmware version.  VLC did *not* work with the TrackMix cameras I had, same stuttering as go2rtc/frigate/ffmpeg direct.  However I have noticed that is the case for a number of other cameras I have tried, such as sv3c.

I have wondered why VLC is sometimes able to recover with only a brief flicker from RTSP streaming errors while it causes the Frigate live/birdeye view to stop streaming and go black/display an error message (with associated stream errors in the log).  However, when you select a specific camera and look at the live feed and history (or look at the feed link within go2rtc) it's usually pretty much intact as well.  The longer Frigate is running without a restart the more likely it is that two of my cameras will be in that state.  I have no way of reliably reproducing the condition so I've never created an issue for that specifically, but it is a problem if you need real time streaming view of all cameras on a monitor and there are cameras that do not follow the RTSP protocol exactly.

**Reply by Waank1** [reply]
The two I have are Firmware: v3.0.0.4255_2411271222 Hardware: IPC_529SD78MP.  Not sure about any other models, but mine have been running fine using RTSP.  I don't use go2rtc but I do use ONVIF.

**Reply by phedders** [reply]
> The two I have are Firmware: v3.0.0.4255_2411271222 Hardware: IPC_529SD78MP. Not sure about any other models, but mine have been running fine using RTSP. I don't use go2rtc but I do use ONVIF.

Would you mind sharing your config for those? Many thanks!

**Reply by Waank1** [reply]
Sure, here it is.  I mainly use this camera to track critters in my back yard around the garden and pool.  I have automations the keep the detection low during the day when wind is blowing the trees around and higher after sunset.

  Critters:
    enabled: true
    ffmpeg:
      inputs:
        - path: rtsp://user:password@192.168.1.249:554/h264Preview_01_sub
          roles:
            - detect
        - path: 
            rtsp://user:password@192.168.1.249:554/h264Preview_01_main
          roles:
            - record
    onvif:
      host: 192.168.1.249
      port: 8000
      user: user
      password: pasword
      autotracking:
        enabled: true
        calibrate_on_startup: true
        track:
          - person
          - cat
          - raccoon
          - deer

        required_zones:
          - Critter_Zone
    record:
      enabled: true
      retain:
        days: 30
        mode: active_objects
      detections:
        pre_capture: 5
        post_capture: 10

    detect:
      width: 3200
      height: 1800
      fps: 5

    snapshots:
      enabled: true
      timestamp: true
      bounding_box: true

    zones:
      Critter_Zone:
        coordinates: 
          -0.001,0.391,0.349,0.438,0.469,0.301,0.552,0.232,0.763,0.349,0.839,0.41,0.993,0.4,0.996,0.995,0.352,0.991,0.309,0.533,0.043,0.559
        loitering_time: 0
    motion:
      mask:
        - 0,0.645,0.063,0.652,0.063,0.713,0.066,1,0,1
        - 0.545,0.23,0.55,0,1,0.002,1,0.394,0.85,0.407,0.586,0.336
        - 0.547,0.228,0.346,0.434,0,0.384,0.002,0,0.55,0.002

**Reply by phedders** [reply]
Interesting - I thought using go2rtc was almost mandatory with >4k Reolinks.
I'm going be giving some of those settings a trial run.

**Reply by Waank1** [reply]
I have some cameras <4k and some >4k.  I have never used go2rtc, and have never used H265.

**Reply by NickM-27** [reply]
The reolink 4K cameras (except for 2 select models) force H265 on the main stream

**Reply by Waank1** [reply]
Nick, TraxMix does not.

**Reply by hagak** [reply]
Trackmix are only H265 on the main stream, I just confirmed this with VLC.

Also Waank1 in your config you have autotracking: enabled: true .  I did not think that works with the reolinks?  Does it actually work for you or do you get the error in the logs that it is ignoring it?

**Comment by JoshuaSeidel** [comment]
use BCS stream or rtsp Preview_Main_02... Sub_02

im using one and have no issues with latest firmware and h265.

i do have this set tho

``` 
ffmpeg:
      hwaccel_args: preset-nvidia-h265
```

**Reply by denisjoshua** [reply]
Can you post your config file please ? 
I have same cam but I didn't get the audio. 
Thanks a lot

**Comment by rpm5099** [comment]
If you really have no stuttering at all then I guess it is just the trackmix wfi.  That certainly is a bit odd considering how similar these devices are.  The only real difference seems to be a wifi radio added into the housing.

**Comment by husnainali09** [comment]
@rpm5099 Did you find a solution. Having this issue as well.

**Comment by Waank1** [comment]
Found this thread today while I was searching for the same problem.  I have had one Trackmix (WIFI version but I use it on LAN), I have it on the corner of my house for about 6 months with no problems at all.  I have it set up in Frigate and every night it records action of the critters that roam around at night.  During the day it also never misses.  I have never had a stutter, skip or dropout and was very happy with it.  BTW I have a total of 10 cameras ranging from very old (2014) to the new Trackmix the only 2 that ever give me problems are the 2 that I have on WIFI).  This past weekend I got another exact same camera on Prime Days.  I set it up exactly as the first cam, with the exception of Zones etc.  After a while I  noticed that not only the new Trackmix was flickering a green screen the original was doing it also.  After some reading and cussing and reading some more I have it fixed.  All I did was change the fps on detect from 10 to 5.  It appears to be fixed, no flicker or green screen for the past 4 hours.  I will ad more if I learn more..  Just how someone on here benefits from this.

Edit: been running now for 28 hours with no problems.

**Reply by rpm5099** [reply]
This is not the first time I have heard of people successfully using the Trackmix with Frigate/RTSP NVR's.  Although if you are running it with H264 you must be using custom firmware I linked above, which looked terrible and unusable on the ones I had.  It could be that I just got two defective trackmix cameras, or it could be that there was an issue in the older firmware/hardware that they have since fixed (and obviously have not disclosed or provided refunds to all of the people they screwed by selling them defective cameras).

**Reply by peter-d-w** [reply]
@Waank1 Could you share your Frigate config file for the Trackmix please? I'm really struggling to get my POE version to work and I'm hoping that your file will help me discover what I'm doing wrong so that I can sort it. I'm very new to Frigate, Reolink & Home Assistant. Thanks

**Reply by Waank1** [reply]
Here ya go Peter, disregard the masks and zones and of course use your IP Address instead of X.XXX.

```
  Critters:
    enabled: true
    ffmpeg:
      inputs:
        - path: rtsp://USER:PASSWORD@192.168.X.XX:554/h264Preview_01_sub
          roles:
            - detect
        - path: 
            rtsp://admin:waank1critters@192.168.1.249:554/h264Preview_01_main
          roles:
            - record
    onvif:
      host: 192.168.1.249
      port: 8000
      user: USER
      password: PASS

      autotracking:
        enabled: true
        calibrate_on_startup: true
        track:
          - person
          - cat
          - raccoon
          - deer

        required_zones:
          - Critter_Zone
    record:
      enabled: true
      retain:
        days: 30
        mode: active_objects
      detections:
        pre_capture: 5
        post_capture: 10

    detect:
      width: 3200
      height: 1800
      fps: 5

    snapshots:
      enabled: true
      timestamp: true
      bounding_box: true

    zones:
      Critter_Zone:
        coordinates: 
          0.318,0.438,0.353,0.438,0.473,0.301,0.548,0.24,0.807,0.216,0.997,0.4,1,0.995,0.338,1,0.301,1,0.246,1
        loitering_time: 0

        inertia: 3
    motion:
      mask:
        - 0,0.391,0.288,0.307,0.324,0.315,0.273,0.753,0.24,0.818,0,1
        - 0.545,0.23,0.55,0,1,0.002,1,0.394,0.849,0.269,0.806,0.21
        - 0.547,0.228,0.346,0.434,0,0.384,0.002,0,0.55,0.002
        - 0.341,0.061,0.32,0.277,0.418,0.315,0.469,0.171,0.421,0.092
      threshold: 30
      contour_area: 30
      improve_contrast: true

**Reply by peter-d-w** [reply]
Thanks, that's great. I appreciate it! Do I need to setup authentication for onvif? If so then how do I do it please? I've turned it on in the settings but I don't see a user name and password field.

**Reply by Waank1** [reply]
I have forgotten how I did it, I think you have to go to Network >Advanced in the Reolink app and choose ONVIF then it asks you to set it up.

**Reply by bdelima** [reply]
@Waank1 I noticed you have the camera set up for autotracking in Frigate. I didn't think the TrackMix supported ONVIF relative movement. Don't you get an error when you start up frigate complaining about the camera?

**Reply by ckruijntjens** [reply]
> Here ya go Peter, disregard the masks and zones and of course use your IP Address instead of X.XXX.
> 
> ```
>   Critters:
>     enabled: true
>     ffmpeg:
>       inputs:
>         - path: rtsp://USER:PASSWORD@192.168.X.XX:554/h264Preview_01_sub
>           roles:
>             - detect
>         - path: 
>             rtsp://admin:waank1critters@192.168.1.249:554/h264Preview_01_main
>           roles:
>             - record
>     onvif:
>       host: 192.168.1.249
>       port: 8000
>       user: USER
>       password: PASS
> 
>       autotracking:
>         enabled: true
>         calibrate_on_startup: true
>         track:
>           - person
>           - cat
>           - raccoon
>           - deer
> 
>         required_zones:
>           - Critter_Zone
>     record:
>       enabled: true
>       retain:
>         days: 30
>         mode: active_objects
>       detections:
>         pre_capture: 5
>         post_capture: 10
> 
>     detect:
>       width: 3200
>       height: 1800
>       fps: 5
> 
>     snapshots:
>       enabled: true
>       timestamp: true
>       bounding_box: true
> 
>     zones:
>       Critter_Zone:
>         coordinates: 
>           0.318,0.438,0.353,0.438,0.473,0.301,0.548,0.24,0.807,0.216,0.997,0.4,1,0.995,0.338,1,0.301,1,0.246,1
>         loitering_time: 0
> 
>         inertia: 3
>     motion:
>       mask:
>         - 0,0.391,0.288,0.307,0.324,0.315,0.273,0.753,0.24,0.818,0,1
>         - 0.545,0.23,0.55,0,1,0.002,1,0.394,0.849,0.269,0.806,0.21
>         - 0.547,0.228,0.346,0.434,0,0.384,0.002,0,0.55,0.002
>         - 0.341,0.061,0.32,0.277,0.418,0.315,0.469,0.171,0.421,0.092
>       threshold: 30
>       contour_area: 30
>       improve_contrast: true 
> ```

Hi,

Are you not using audio? if not using web2rtc audio is not possible right?

**Reply by Waank1** [reply]
> @Waank1 I noticed you have the camera set up for auto tracking in Frigate. I didn't think the TrackMix supported ONVIF relative movement. Don't you get an error when you start up frigate complaining about the camera?

[bdelima](https://github.com/bdelima). You are correct.  I had that in there from the beginning, and have since learned a bit more and have commented it out.  I found out the the Reolink software takes care of all that.  I can move the camera(s) manually using the (custom:advanced-camera-card), and send them back to the 'home' position, the only thing I have not been able to do is use the camera(s) zoom, just have to scroll manually with the mouse wheel to digitally get closer.  I will edit my earlier post to correct this.  Thanks,

**Reply by HubertusH** [reply]
you can use zoom from frigate if you use the stream h264Preview_**02**_main

**Reply by rpm5099** [reply]
> Found this thread today while I was searching for the same problem. I have had one Trackmix (WIFI version but I use it on LAN), I have it on the corner of my house for about 6 months with no problems at all. I have it set up in Frigate and every night it records action of the critters that roam around at night. During the day it also never misses. I have never had a stutter, skip or dropout and was very happy with it. BTW I have a total of 10 cameras ranging from very old (2014) to the new Trackmix the only 2 that ever give me problems are the 2 that I have on WIFI). This past weekend I got another exact same camera on Prime Days. I set it up exactly as the first cam, with the exception of Zones etc. After a while I noticed that not only the new Trackmix was flickering a green screen the original was doing it also. After some reading and cussing and reading some more I have it fixed. All I did was change the fps on detect from 10 to 5. It appears to be fixed, no flicker or green screen for the past 4 hours. I will ad more if I learn more.. Just how someone on here benefits from this.
> 
> Edit: been running now for 28 hours with no problems.

@Waank1 I think Reolink made an attempt to fix the RTSP issues and I have been hearing that later versions of the trackmix are working ok.  I'm glad it worked out for you.  If you run into issues/resolutions in the future please let us know.

**Comment by BoutTime2** [comment]
Just wondering if any of you still have the Reolink software trying to auto detect, auto track and auto zoom at the same time and therefore fight with Frigate for bandwidth, memory and CPU resources?

**Reply by HubertusH** [reply]
Autotrack did not work from frigate to TrackMix - only the autotrack of the cam himself works.

**Comment by HubertusH** [comment]
I also had extreme problems with Track Mix. My other Reolink camera, the RLC-823A 16X, works perfectly with the default settings in Frigate.

For Track Mix, I set Preview_02_main (or _sub) instead of Preview_01_main (or _sub), and since then I haven't had any error messages, and ONIV now also works with zoom.

**Comment by ckruijntjens** [comment]
I still have the stuttering in frigate with this camera. What you guys have as full configs?????

**Comment by ckruijntjens** [comment]
and what settings on the camera?

**Comment by HubertusH** [comment]
I'm sorry, I can't give any more answers on that—the TrackMix has given up the ghost, and I've now installed a different camera.

**Comment by fotingo** [comment]
I use the Trackmix POE in Blue Iris and have zero issues. It works very well.

**Comment by beatles1235** [comment]
I have a few Trackmix PoE cameras running, it's been a painful process, but the most stable firmware I received from Reolink, similar to the op, just for the PoE model though

https://drive.google.com/file/d/1o3-6se5Fg5ZYhZD_uUTJammaWXvJ1MiL/view?usp=drive_link
go2rtc:
  preload:                       
 
  streams:
    Front_Driveway: # <- for RTSP streams
      - rtsp://user:pass@10.1.1.120:554/h264Preview_01_main
      - rtsp://user:pass@10.1.1.120:554/h264Preview_01_sub #for two-way audio
      - ffmpeg:Front_Driveway#audio=opus#video=h264#hardware  

  Front_Driveway:
    detect:
      width: 1920
      height: 1080
      fps: 1

    ffmpeg:
      output_args:
        record: preset-record-generic-audio-copy
      inputs:
        - path: rtsp://127.0.0.1:8554/Front_Driveway
          input_args: preset-rtsp-restream
          roles:
            - record
            - detect
            - audio
    live:
      streams:
        Front_Driveway: Front_Driveway


Pretty stable for me.

**Reply by phedders** [reply]
Thankyou - will give that a try ASAP.

**Reply by AdamS3x** [reply]
@beatles1235 Would you say it's worth it to run these cameras give your experience? I'm looking at getting a few Trackmix PoE's for a brand new first time setup with HA/Frigate and I really want this camera for it's capabilities, but does everything work properly in Frigate i.e. 2 way audio, detection, control and no image issues now you've gone through this painful process and got a stable config that works? Interested in the experience you've had if you care to share. Thanks.

**Comment by beatles1235** [comment]
With the firmware version I posted, yes. I did want to swap to Ubiquitis
PTZ but for the cost, it’s pretty good. Yes, everything works (audio, no
frame issues) and it’s pretty stable.

Personally I wouldn’t buy any more, but it was hard to justify buying new
cameras when these work. The only thing that doesn’t work is the auto
tracking within frigate, but it does work natively with the cameras
settings, so it’ll track people, pets and cars. I personally don’t really
like using the tracking since frigate has no idea, so you’ll get car
notifications if a person walks by. Overall though I’m happy with it

**Reply by AdamS3x** [reply]
> With the firmware version I posted, yes. I did want to swap to Ubiquitis PTZ but for the cost, it’s pretty good. Yes, everything works (audio, no frame issues) and it’s pretty stable. Personally I wouldn’t buy any more, but it was hard to justify buying new cameras when these work. The only thing that doesn’t work is the auto tracking within frigate, but it does work natively with the cameras settings, so it’ll track people, pets and cars. I personally don’t really like using the tracking since frigate has no idea, so you’ll get car notifications if a person walks by. Overall though I’m happy with it
> […](#)
> On Sun, Mar 22, 2026 at 2:10 PM AdamS3x ***@***.***> wrote: @beatles1235 <https://github.com/beatles1235> Would you say it's worth it to run these cameras give your experience? I'm looking at getting a few Trackmix PoE's for a brand new first time setup with HA/Frigate and I really want this camera for it's capabilities, but does everything work properly in Frigate i.e. 2 way audio, detection, control and no image issues now you've gone through this painful process and got a stable config that works? Interested in your experience you've had if you care to share. Thanks. — Reply to this email directly, view it on GitHub <#15156?email_source=notifications&email_token=ADY6FSOUKG423FJVZEOYPIL4SAUAHA5CNFSNUABIM5UWIORPF5TWS5BNNB2WEL2ENFZWG5LTONUW63SDN5WW2ZLOOQXTCNRSGYYDMMZXUZZGKYLTN5XKO3LFNZ2GS33OUVSXMZLOOSWGM33PORSXEX3DNRUWG2Y#discussioncomment-16260637>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ADY6FSOLGYOB65KBWPYXWNT4SAUAHAVCNFSM6AAAAABSMARWQWVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTMMRWGA3DGNY> . You are receiving this because you were mentioned.Message ID: ***@***.*** com>

Hmm, so the auto tracking doesn't work in Frigate? That seems like a pretty big problem. Is there no way to get that working?

**Reply by hagak** [reply]
This is the case with MANY PTZ cameras not just this one.  The issue is the camera has to support a set of features that many dont. https://docs.frigate.video/configuration/cameras#onvif-ptz-camera-recommendations

**Comment by beatles1235** [comment]
I believe it’s due to how Reolink implemented it. It doesn’t support relative movement within the field of view. The camera will still move based off what you select in the Reolink app, frigate just will not be aware the camera is working. Sent from my iPhoneOn Mar 28, 2026, at 2:37 PM, AdamS3x ***@***.***> wrote:﻿

With the firmware version I posted, yes. I did want to swap to Ubiquitis PTZ but for the cost, it’s pretty good. Yes, everything works (audio, no frame issues) and it’s pretty stable. Personally I wouldn’t buy any more, but it was hard to justify buying new cameras when these work. The only thing that doesn’t work is the auto tracking within frigate, but it does work natively with the cameras settings, so it’ll track people, pets and cars. I personally don’t really like using the tracking since frigate has no idea, so you’ll get car notifications if a person walks by. Overall though I’m happy with it
…
On Sun, Mar 22, 2026 at 2:10 PM AdamS3x @.> wrote: @beatles1235 https://github.com/beatles1235 Would you say it's worth it to run these cameras give your experience? I'm looking at getting a few Trackmix PoE's for a brand new first time setup with HA/Frigate and I really want this camera for it's capabilities, but does everything work properly in Frigate i.e. 2 way audio, detection, control and no image issues now you've gone through this painful process and got a stable config that works? Interested in your experience you've had if you care to share. Thanks. — Reply to this email directly, view it on GitHub <#15156?email_source=notifications&email_token=ADY6FSOUKG423FJVZEOYPIL4SAUAHA5CNFSNUABIM5UWIORPF5TWS5BNNB2WEL2ENFZWG5LTONUW63SDN5WW2ZLOOQXTCNRSGYYDMMZXUZZGKYLTN5XKO3LFNZ2GS33OUVSXMZLOOSWGM33PORSXEX3DNRUWG2Y#discussioncomment-16260637>, or unsubscribe https://github.com/notifications/unsubscribe-auth/ADY6FSOLGYOB65KBWPYXWNT4SAUAHAVCNFSM6AAAAABSMARWQWVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTMMRWGA3DGNY . You are receiving this because you were mentioned.Message ID: @. com>

Hmm, so the auto tracking doesn't work in Frigate? That seems like a pretty big problem. Is there no way to get that working?

**Comment by pripyaat** [comment]
Since this thread is one of the top search results for "Reolink TrackMix RTSP issues", I wanted to share my findings in case they help someone else.

1. I can also confirm **this camera** (along with a couple of other Reolink models) **does have RTSP-specific issues**. By that, I mean that audio stuttering, video stuttering, image corruption and/or apparent packet loss are exclusively tied to the RTSP stream, while the other protocols (RTMP, HTTP-FLV) don't exhibit the same issues.

2. Although Reolink provides custom firmware that supposedly enables H.264 encoding for the Main stream, that doesn't appear to be entirely accurate. For example, the configuration posted by @beatles1235 is **not** actually ingesting an H.264 stream from the camera, but instead it is transcoding the video on the fly using FFmpeg with hardware acceleration: 

    `ffmpeg:Front_Driveway#audio=opus#video=h264#hardware.` 
    
    Even though the URL contains **h264Preview_01_main**, in my testing the camera is _still delivering H.265 (HEVC) video_.

3. Since the Main stream is encoded in H.265, applications using **FFmpeg 7.x or earlier** cannot properly decode the RTMP or HTTP-FLV stream. However, the good news is that [this documentation](https://support.reolink.com/articles/28256840140441-Introduction-to-FLV-Stream/) appears to be outdated, as **FFMPEG 8.0 and newer can actually decode the H.265 HTTP-FLV stream**. While this is not a fix for the RTSP issues themselves, it does provide an excellent workaround.

So, if you're using Frigate 0.18 (which is already bundled with FFMPEG 8.0), you can simply use the following configuration. It should work flawlessly without the need for custom firmware or video transcoding:

```
streams:
    trackmix_main:
      - "ffmpeg:http://xxx.xxx.xxx.xxx/flv?port=1935&app=bcs&stream=channel0_main.bcs&user=admin&password=password#video=copy#audio=copy#audio=aac"
```
_Note:_ **the HTTP and RTMP ports must be enabled** on the camera for this to work. If you want to use the stream over HTTPS and you're using a self-signed or untrusted certificate, you need to add the following FFmpeg input argument: -tls_verify 0

EDIT: Updated the last part since it's no longer needed to use a custom FFMPEG path.

**Comment by baudneo** [comment]
I don't own one of these *yet*, but there is the possibility of using a custom built go2rtc binary that extracts the raw H265/264 full resolution stream and wraps it so frigate can ingest it (no transcoding). It interfaces using the native Baichuan protocol via port 9000. The repo its implemented in is: https://github.com/ucbmckee/go2rtc-ucbmckee/tree/topic/reolink-baichuan

You can view the commit log to see the pertinent README parts (even says it was tested using a TrackMix camera). Here is a docker one-liner to spit out an amd64 binary you can place in the frigate `/config` dir:

```
docker run --rm -v "$(pwd):/out" golang:1.24-alpine sh -c "
  apk add --no-cache git &&
  git clone -b topic/reolink-baichuan https://github.com/ucbmckee/go2rtc-ucbmckee.git /src &&
  cd /src &&
  CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -ldflags '-s -w' -trimpath -o /out/go2rtc .
"
```

Example go2rtc config:

```
      - reolink://USER:PASS@IPADDR/main
      - reolink://USER:PASS@IPADDR/extern
```

I stumbled upon this repo when diagnosing reolink RTSP two way audio delays. It seems it would be useful here as well.
